src/calculator/index.ts

```diff
@@ -266,10 +277,138 @@ export default class Calculator {
     return augmented;
   }
 
+  /**
+   * Estimates matching for a given project and potential additional votes
+   * @param potentialVotes
+   * @param roundId
+   */
+  async estimateMatching(
+    potentialVotes: PotentialVote[],
+    roundId: string
+  ): Promise<MatchingEstimateResult[]> {
+    const votes = await this.parseJSONFile<Vote>(
+      "votes",
+      `${this.chainId}/rounds/${this.roundId}/votes.json`
+    );
+
+    const rounds = await this.parseJSONFile<Round>(
+      "rounds",
+      `${this.chainId}/rounds.json`
+    );
+
+    const round = rounds.find((r: Round) => r.id === this.roundId);
+
+    if (round === undefined) {
+      throw new ResourceNotFoundError("round");
+    }
+
+    if (round.matchAmount === undefined) {
+      throw new ResourceNotFoundError("round match amount");
+    }
+
+    if (round.token === undefined) {
+      throw new ResourceNotFoundError("round token");
+    }
+
+    const applications = await this.parseJSONFile<Application>(
+      "applications",
+      `${this.chainId}/rounds/${this.roundId}/applications.json`
+    );
+
+    const currentResults = await this._calculate(votes);
+    const potentialVotesAugmented: Vote[] = await Promise.all(
+      potentialVotes.map(async (vote) => {
+        const { amount: amountUSD } = await this.priceProvider.convertToUSD(
```

Comment:
Just something to think about, but a`getUsdConversionRate(chainId, token)` in the price provider would allow to make one async call only outside of this loop instead of `potentialVotes.length * 3` async calls. /cc @boudra

Reply:
Yes, we should fix this, we had a similar problem in the past where we were looping through all the Passports for each vote. There's already a `getUSDConversionRate`, it looks like the process of then converting the `bigint` amount to USD (floating point) is not straightforward, but that can be extracted into it's own function.

Reply:
I agree, we can add it directly there or having it here getting a price with a constant amount to calculate the rate.

Reply:
Good idea - fixed.